streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -157,26 +165,82 @@ public void shouldIdempotentlyUpdateSubscriptionFromActiveAssignment() {
     }
 
     @Test
-    public void shouldReturnOffsetsForAllCachedTaskIdsFromDirectory() throws IOException {
-        final File[] taskFolders = asList(testFolder.newFolder("0_1"),
-                                          testFolder.newFolder("0_2"),
-                                          testFolder.newFolder("0_3"),
-                                          testFolder.newFolder("1_1"),
-                                          testFolder.newFolder("dummy")).toArray(new File[0]);
+    public void shouldReportOffsetSumsForValidLockedTasks() throws IOException {
+        final File[] taskFolders = asList(/* SHOULD report offsets for the following cases: */
+            testFolder.newFolder("0_0"),     // active running task
+            testFolder.newFolder("0_1"),     // active non-running task
+            testFolder.newFolder("0_2"),     // standby task
+            testFolder.newFolder("1_0"),     // unowned (unlocked) task with valid checkpoint
+                                          /* should NOT report offsets for the following: */
+            testFolder.newFolder("1_1"),     // unowned (unlocked) task without checkpoint file
+            testFolder.newFolder("1_2"),     // owned/locked by another thread
+            testFolder.newFolder("dummy"))   // some random non-task dir
+                                       .toArray(new File[0]);
+
+        final Map<TopicPartition, Long> task01ChangelogOffsets = mkMap(
+            mkEntry(t1p0, 1L),
+            mkEntry(t1p1, 2L)
+        );
+        final Map<TopicPartition, Long> task02ChangelogOffsets = mkMap(
+            mkEntry(t1p0, 5L),
+            mkEntry(t1p1, 10L)
+        );
+        final Map<TopicPartition, Long> task10ChangelogOffsets = mkMap(
+            mkEntry(t1p0, 20L),
+            mkEntry(t1p1, 30L)
+        );
```

Comment:
Yeah, I was just reusing them to be lazy and because they don't matter. But better to avoid confusion with separate variables/names.

Agreed on varying the partition count,  will also add some "special cases" like -1